themes/V3/5ePHB/snippets/tableOfContents.gen.js

```diff
@@ -22,37 +22,39 @@ const getTOC = (pages)=>{
 		if(!_.last(_.last(res).children)) add2(null, page);
 		_.last(_.last(res).children).children.push({
 			title    : title,
-			page     : page + 1,
+			page     : page,
 			children : []
 		});
 	};
 
 	const res = [];
-	_.each(pages, (page, pageNum)=>{
-		if(!page.includes("{{frontCover}}") && !page.includes("{{insideCover}}") && !page.includes("{{partCover}}") && !page.includes("{{backCover}}")) {
-			const lines = page.split('\n');
-			_.each(lines, (line)=>{
-				if(_.startsWith(line, '# ')){
-					const title = line.replace('# ', '');
-					add1(title, pageNum);
+	const iframe = document.getElementById('BrewRenderer');
+	const iframeDocument = iframe.contentDocument || iframe.contentWindow.document;
+	const headings = iframeDocument.querySelectorAll('h1, h2, h3');
+
+	_.each(headings, (heading)=>{
+		const onPage = parseInt(heading.closest('.page,.phb').id?.replace(/^p/, ''));
+		const ToCExclude = getComputedStyle(heading).getPropertyValue('--TOC');
+		if(ToCExclude != 'exclude') {
+			if(!isNaN(onPage)) {
+				const headingText =  heading.innerText;
+				if(heading.tagName == 'H1') {
+					add1(headingText, onPage);
 				}
-				if(_.startsWith(line, '## ')){
-					const title = line.replace('## ', '');
-					add2(title, pageNum);
+				if(heading.tagName == 'H2') {
+					add2(headingText, onPage);
 				}
-				if(_.startsWith(line, '### ')){
-					const title = line.replace('### ', '');
-					add3(title, pageNum);
+				if(heading.tagName == 'H3') {
```

Comment:
We need cases for calls to `add4` and `add5` here.

Reply:
I added h4, h5, and h6 here. I seem to have missed when in this timeline we added H6, but there you go.